packages/@aws-cdk/cloudformation-include/test/nested-stacks.test.ts

```diff
@@ -0,0 +1,585 @@
+    expect(() => {
+      new inc.CfnInclude(stack, 'ParentStack', {
+        templateFile: testTemplateFilePath('parent-creation-policy.json'),
+        nestedStacks: {
+          'ChildStack': {
+            templateFile: testTemplateFilePath('grandchild-import-stack.json'),
+          },
+        },
+      });
+    }).toThrow(/CreationPolicy is not supported by the AWS::CloudFormation::Stack resource/);
+  });
+
+  test('Throws an error when updatePolicy is defined', () => {
+    expect(() => {
+      new inc.CfnInclude(stack, 'ParentStack', {
+        templateFile: testTemplateFilePath('parent-update-policy.json'),
+        nestedStacks: {
+          'ChildStack': {
+            templateFile: testTemplateFilePath('grandchild-import-stack.json'),
+          },
+        },
+      });
+    }).toThrow(/UpdatePolicy is not supported by the AWS::CloudFormation::Stack resource/);
+  });
+
+  test('Throws an error when referring to an undefined condition', () => {
+    expect(() => {
+      new inc.CfnInclude(stack, 'ParentStack', {
+        templateFile: testTemplateFilePath('parent-invalid-condition.json'),
+        nestedStacks: {
+          'ChildStack': {
+            templateFile: testTemplateFilePath('grandchild-import-stack.json'),
+          },
+        },
+      });
+    }).toThrow(/nested stack 'ChildStack' uses Condition 'FakeCondition' that doesn't exist/);
+  });
+
+  test('Throws an error when a nested stacks depends on an undefined resource', () => {
+    expect(() => {
+      new inc.CfnInclude(stack, 'ParentStack', {
+        templateFile: testTemplateFilePath('parent-bad-depends-on.json'),
+        nestedStacks: {
+          'ChildStack': {
+            templateFile: testTemplateFilePath('child-import-stack.json'),
+          },
+        },
+      });
+    }).toThrow(/nested stack 'ChildStack' depends on 'AFakeResource' that doesn't exist/);
+  });
+
+  test('can modify resources in nested stacks', () => {
+    const parent = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('child-import-stack.json'),
+      nestedStacks: {
+        'GrandChildStack': {
+          templateFile: testTemplateFilePath('grandchild-import-stack.json'),
+        },
+      },
+    });
+
+    const childTemplate = parent.getNestedStack('GrandChildStack').includedTemplate;
+    const bucket = childTemplate.getResource('BucketImport') as s3.CfnBucket;
+
+    bucket.bucketName = 'modified-bucket-name';
+
+    expect(childTemplate.stack).toHaveResource('AWS::S3::Bucket', { BucketName: 'modified-bucket-name' });
+  });
+
+  test('can use a condition', () => {
+    const parent = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-valid-condition.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('grandchild-import-stack.json'),
+        },
+      },
+    });
+
+    const AlwaysFalseCondition = parent.getCondition('AlwaysFalseCond');
```

Comment:
```suggestion
    const alwaysFalseCondition = parent.getCondition('AlwaysFalseCond');
```

Variables are lowercase.